geoprocessing/src/modules/planning-units/planning-units.job.ts

```diff
@@ -0,0 +1,72 @@
+import { Logger } from '@nestjs/common';
+import { Job } from 'bullmq';
+import { geoprocessingConnections } from 'src/ormconfig';
+import { createConnection } from 'typeorm';
+
+const logger = new Logger('planning-units-job-processor');
+
+export enum PlanningUnitGridShape {
+  square = 'square',
+  hexagon = 'hexagon',
+  fromShapefile = 'from_shapefile',
+}
+export interface PlanningUnitsJob {
+  countryId?: string;
+  adminRegionId?: string;
+  adminAreaLevel1Id?: string;
+  adminAreaLevel2Id?: string;
+  planningUnitGridShape: PlanningUnitGridShape;
+  planningUnitAreakm2: number;
+  extent?: Record<string, unknown>;
+}
+
+export default async (job: Job<PlanningUnitsJob> | any) => {
```

Comment:
```suggestion
export const createPlanningUnitGridFromJobSpec = async (job: Pick<Job<PlanningUnitsJob>, 'data' | 'id' | 'name'>) => {
```
just a tiny detail, but I recommend to name exports explicitly to help understand at a glance what they are - this suggestion is mainly about the type signature (see comments below for `planning-units-processor.e2e-spec.ts`) but I also suggest the naming thingy (and use that in the e2e-spec file suggestions)

Reply:
sorry, this is confusing, I just realized this now: it was an initial suggestion, then the PR review page froze and when I reloaded it this comment wasn't shown, and in the meanwhile I had checked it locally and realised that naming the export was leading to the function not being called, so I left this alone thinking that the comment had been lost, but no, the part about naming the export should be ignored (and the part about the type signature of the job parameter could be considered, if you want). sorry about the confusion!

Reply:
thank you for this :D